cirq/devices/graph_device.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import itertools
+
+from typing import cast, FrozenSet, Hashable, Iterable, Optional
+
+from cirq import circuits, devices, ops, value, schedules
+
+from cirq.devices.hypergraph import UndirectedHypergraph
+
+
+class HashQubit(ops.QubitId):
```

Comment:
This definition is out of place here. Qubits are already hashable. I would just delete this class from the PR.

Reply:
The name is terrible, but I think this is needed. The "hash" refers to the `value` method being hashable, not the qubit itself. The idea is something like `NamedQubit`, but where the analog to `name` is not restricted to being a string.

Reply:
What about `VertexQubit`?

Reply:
`QubitId` already requires hashability. I think we should just use `QubitId`. This class just isn't pulling its weight in terms of added value; it's an extra detail that the user must remember when instead you could just say "any qubit works".

Reply:
Done.